backend/src/api/paths/submissions.ts

```diff
@@ -62,21 +79,66 @@ const getSubmissionByIdRoute = createRoute({
 })
 
 // ルートの設定
-app.openapi(getSubmissionsRoute, (c) => {
-  // TODO: 実際のデータベースクエリを実装
-  const submissions: z.infer<typeof Submission>[] = [
-    {
-      code: "print('Hello, World!')",
-      id: 1,
-      language: { name: "Python", version: "3.9" },
-      problem_id: 1,
-      result: { message: "テストケースにパスしました", status: "Accepted" },
-      student_id: 1,
-      submitted_at: new Date().toISOString(),
-      test_results: [{ message: "正解", status: "Passed", test_case_id: 1 }],
+app.openapi(getSubmissionsRoute, async (c) => {
+  const submissions = await prisma.submission.findMany({
+    include: {
+      language: true,
+      problem: true,
+      result: {
+        include: {
+          status: true,
+        },
+      },
+      testResults: {
+        include: {
+          status: true,
+          testCase: true,
+        },
+      },
     },
-  ]
-  return c.json(submissions)
+    orderBy: {
+      createdAt: "desc",
+    },
+  })
+
+  const formattedSubmissions = submissions.map((submission) => {
+    const submissionStatus = submission.result.status.status
+    if (!isValidSubmissionStatus(submissionStatus)) {
+      throw new Error(`Invalid submission status: ${submissionStatus}`)
+    }
+
+    const formattedTestResults = submission.testResults.map((result) => {
+      const testStatus = result.status.status
+      if (!isValidTestStatus(testStatus)) {
+        throw new Error(`Invalid test status: ${testStatus}`)
+      }
+
+      return {
+        message: result.message,
+        status: testStatus,
+        test_case_id: result.testCaseId,
+      } as z.infer<typeof TestResult>
+    })
+
+    return {
+      code: submission.code,
+      id: submission.id,
+      language: {
+        name: submission.languageName,
+        version: submission.languageVersion,
+      },
+      problem_id: submission.problemId,
+      result: {
+        message: submission.result.message,
+        status: submissionStatus,
+      },
+      student_id: submission.studentId,
+      submitted_at: submission.createdAt.toISOString(),
+      test_results: formattedTestResults,
+    } as z.infer<typeof Submission>
```

Comment:
`as`だと、オブジェクトの中身が`z.infer<typeof Submission>`と一致しなくても型情報を上書きしてしまうので、`satisfies`位にしておいた方が良いと思われます
もしくは、オブジェクトを変数として定義して、それに対して型アノテーションを与えるのが良いと思います

```suggestion
    } satisfies z.infer<typeof Submission>
```